### PR TITLE
Furthering the work on EA record creation/service communication

### PR DESCRIPTION
**Pull request by borkweb**
Here's the current progress on the creation of Import Records. I've begun the Facebook Record while building out the Record Abstract class, logging of errors as comments, handling daily Event Aggregator request limits, and the queueing of imports.

Loosely see: https://central.tri.be/issues/61161

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Don't leave debug_robot lying around | [Aggregator.php:213](https://github.com/moderntribe/the-events-calendar/blob/484b238282e4ee74ea53fd37df067170b1a9eeaf/src/Tribe/Aggregator.php#L213)
:x: | TRUE, FALSE and NULL must be lowercase; expected &quot;true&quot; but found &quot;TRUE&quot; | [Aggregator.php:213](https://github.com/moderntribe/the-events-calendar/blob/484b238282e4ee74ea53fd37df067170b1a9eeaf/src/Tribe/Aggregator.php#L213)
:x: | Don't leave debug_robot lying around | [Page.php:256](https://github.com/moderntribe/the-events-calendar/blob/484b238282e4ee74ea53fd37df067170b1a9eeaf/src/Tribe/Aggregator/Page.php#L256)
:x: | TRUE, FALSE and NULL must be lowercase; expected &quot;true&quot; but found &quot;TRUE&quot; | [Page.php:256](https://github.com/moderntribe/the-events-calendar/blob/484b238282e4ee74ea53fd37df067170b1a9eeaf/src/Tribe/Aggregator/Page.php#L256)
:x: | Don't leave debug_robot lying around | [Page.php:263](https://github.com/moderntribe/the-events-calendar/blob/484b238282e4ee74ea53fd37df067170b1a9eeaf/src/Tribe/Aggregator/Page.php#L263)
:x: | TRUE, FALSE and NULL must be lowercase; expected &quot;true&quot; but found &quot;TRUE&quot; | [Page.php:263](https://github.com/moderntribe/the-events-calendar/blob/484b238282e4ee74ea53fd37df067170b1a9eeaf/src/Tribe/Aggregator/Page.php#L263)
:x: | Don't leave debug_robot lying around | [Abstract.php:205](https://github.com/moderntribe/the-events-calendar/blob/484b238282e4ee74ea53fd37df067170b1a9eeaf/src/Tribe/Aggregator/Record/Abstract.php#L205)
:x: | TRUE, FALSE and NULL must be lowercase; expected &quot;true&quot; but found &quot;TRUE&quot; | [Abstract.php:205](https://github.com/moderntribe/the-events-calendar/blob/484b238282e4ee74ea53fd37df067170b1a9eeaf/src/Tribe/Aggregator/Record/Abstract.php#L205)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Comment refers to a TODO task &quot;do something with the submission of events. $_POST['data'&quot; | [Aggregator.php:116](https://github.com/moderntribe/the-events-calendar/blob/fbaeb8240e9335a5bc1730ff0016e2e3d6ecd5bc/src/Tribe/Aggregator.php#L116)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Class name is not valid; consider Tribe_Events_Aggregator_Record_ICal instead | [iCal.php:3](https://github.com/moderntribe/the-events-calendar/blob/8724b9feeda9ad68121488f220be067597980c90/src/Tribe/Aggregator/Record/iCal.php#L3)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Usage of &quot;$this&quot; in static methods will cause runtime errors | [Factory.php:50](https://github.com/moderntribe/the-events-calendar/blob/70e4b8ea51b16d1acc113a7b9987a449ad2a16fd/src/Tribe/Aggregator/Record/Factory.php#L50)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Aggregator.php [hunk 1]
```diff
@@ -45,23 +52,13 @@ public static function instance() {
...
 	 */
 	public function __construct() {
-		$this->page        = Tribe__Events__Aggregator__Page::instance();
-		$this->service     = Tribe__Events__Aggregator__Service::instance();
-		$this->record      = Tribe__Events__Aggregator__Record::instance();
-		$this->cron        = Tribe__Events__Aggregator__Cron::instance();
-		$this->pue_checker = new Tribe__PUE__Checker( 'http://tri.be/', 'event-aggregator' );
+		$this->page             = Tribe__Events__Aggregator__Page::instance();
+		$this->service          = Tribe__Events__Aggregator__Service::instance();
+		$this->record_post_type = Tribe__Events__Aggregator__Record__Post_Type::instance();
```
Comment (bordoni) [inline src/Tribe/Aggregator.php]: Can we rename it to `$this->records`?

File: src/Tribe/Aggregator.php [hunk 2]
```diff
@@ -150,4 +155,81 @@ public function api( $api = null ) {
...
+	public function daily_limit() {
+		return $this->daily_limit;
+	}
+
+	/**
+	 * Returns the available daily limit of import requests
+	 *
+	 * @return int
+	 */
+	public function daily_limit_available() {
```
Comment (bordoni) [inline src/Tribe/Aggregator.php]: Shouldn't this be name like: `get_daily_limit`?

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [refine.php:19](https://github.com/moderntribe/the-events-calendar/blob/07464af52ab34e0384b9c84b15a86dc7d81596d4/src/admin-views/aggregator/origins/refine.php#L19)

> via jenkins/codesniffer-phpcs

**Comment by borkweb** [comment]
Merging this bad boy!

**Pull request merged**